### PR TITLE
Add manacher algorithm

### PR DESCRIPTION
**Pull request by ulwlu**
Added manacher algorithm.

- [x] cargo fmt
- [x] cargo test

This algorithm can pass the [leetcode's longest palindromic problem](https://leetcode.com/problems/longest-palindromic-substring/) as well.

### CODE REVIEW

File: src/string/manacher.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+#[cfg(test)]
+mod tests {
+    use super::manacher;
+
+    #[test]
+    fn get_longest_palindrome_by_manacher() {
+        assert_eq!(manacher("babad".to_string()), "aba".to_string());
+        assert_eq!(manacher("cbbd".to_string()), "bb".to_string());
+        assert_eq!(manacher("a".to_string()), "a".to_string());
+        assert_eq!(manacher("ac".to_string()), "c".to_string());
```
[the remark this answers is not in the document]
Comment (siriak) [reply src/string/manacher.rs:87]: "c" is 1 character long and "a" is 1 character long. So, they are of the same length and so they are both the longest.

File: src/string/README.md [hunk 2]
```diff
@@ -10,6 +10,14 @@ __Properties__
 
 [kmp-wiki]: https://en.wikipedia.org/wiki/Knuth–Morris–Pratt_algorithm
 
+### [Manacher](./manacher.rs)
+From [Wikipedia][manacher-wiki]: find a longest palindrome in a string in linear time.
+
+__Properties__
+* Case performance  O(n)
+* Case space complexity  O(n)
```
Comment (siriak) [inline src/string/README.md:18]:
```suggestion
* Worst-case time complexity is O(n)
* Worst-case space complexity is O(n)
```

**Review by siriak — CHANGES_REQUESTED** [review]

File: src/string/manacher.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ulwlu) [reply src/string/manacher.rs:87]: I know what you said, but I said we need to get the one and this is supposed to be so. 
If you want to get the all occurance you should set hash to store but it's your usecase.

[the remark this answers is not in the document]
Comment (imp2002) [reply src/string/manacher.rs:87]: manacher usually only returns one not multiple correct answer.

[the remark this answers is not in the document]
Comment (imp2002) [reply src/string/manacher.rs:87]: I think it's ok, in many other repo of algorithm the manacher is just return one too.

**Review by imp2002 — APPROVED** [review]

File: src/string/README.md (hunk 2 above)
Comment (ulwlu) [reply src/string/README.md]: @siriak 
If you suggest changing so, I recommend you changing below as well.

You should keep same format.

https://github.com/TheAlgorithms/Rust/blob/master/src/string/README.md

File: src/string/manacher.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (siriak) [reply src/string/manacher.rs:87]: I don't say that it should return all the answers, I say that it should return any one of them. And returning "a" would be considered incorrect in this test, while in fact, the "a" is also a correct answer.

File: src/string/README.md (hunk 2 above)
Comment (siriak) [reply src/string/README.md]: Agree, let's change it everywhere

File: src/string/manacher.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (imp2002) [reply src/string/manacher.rs:87]: `let longest_palindromic_substring = manacher("ac".to_string());`

`assert!(longest_palindromic_substring == "a".to_string() || longest_palindromic_substring == "c".to_string());`

Is this ok?

[the remark this answers is not in the document]
Comment (siriak) [reply src/string/manacher.rs:87]: Yes, this LGTM

[the remark this answers is not in the document]
Comment (ulwlu) [reply src/string/manacher.rs]: @imp2002 thank you, fixed here.

9a88535

**Review by imp2002 — APPROVED** [review]

**Review by siriak — APPROVED** [review]
Cool, thanks!

**Pull request merged**